2.x feature -> 2.x develop | Updated: permission denied response

we-alam-a001 opened the pull request:
[Task #528 - 2023-12-27 VaahCMS > Update Hint In Permission Denied Respose](https://team.webreinvent.com/backend/collab/project/vaahcms/tasks/528)
Time invested: 11:00 format | Billable:  11:00 | Non-Billable: 00:00

Client's comment or requirement

Proof of your testing, video, or image link?

video link :- (https://img-v1.dev.getdemo.dev/screenshot/chrome_JCZYzGXB6n.mp4
### Self Review Checklist

#### Naming Conventions
> For a detailed information click [here](https://docs.vaah.dev/guide/code.html#naming-conventions)

- [ ] Are your Controller names meaningful, singular and have Controller suffix?
- [ ] Are your Model names singular and named after the table they are modelling?
- [ ] Are your Table names snake_case and plural?
- [ ] Are your Table Column names snake_case without model name?
- [ ] Are your Seeder names descriptive about which table they are seeding?
- [x] Are your method/function names short, descriptive and camelCased?
- [x] Are your variable names short, descriptive and snake_cased?

#### Constants and Configs
> For a detailed information click [here](https://docs.vaah.dev/guide/code.html#naming-conventions)

- [x] API keys, Secret Keys, URL's etc., are stored as env variables.
- [x] I am accessing env variables via config.
- [x] I have used config and language files, constants instead of text in the code .
- [x] I have not hardcoded any value.

#### Fat methods  and CRUD
> For a detailed information click [here](https://docs.vaah.dev/guide/code.html#laravel-community-guidelines-for-good-coding-practices)

- [x] All requests are being validated
- [ ] I am not using raw SQL queries. I am using Eloquent queries.
- [ ] DB related logic are in Eloquent models or in Repository/Helper classes.
- [x] I have not included business logic in Controllers. They are in Repository/Helper classes.
- [ ] My methods adhere to the Single Responsibility Principle.
- [ ] My code doesn't contain extensive if-else nesting. I am checking negative conditions first.
- [x] I have removed all unwanted commented code.
- [ ] I have done my tech debt analysis and any unavoidable code leading to tech debt is marked with @todo .

#### Other practices

- [ ] All my migrations have down method that reverses operations performed by up method.
- [ ] Any new column added/altered in table has been also updated in the fillable property of its model.
- [x] For seeders that are not populating sample data, it is not truncating table before seeding again.
- [ ] For seeders that are not populating sample data, the data is being populated from a separate external file.
- [x] There are no business logic or queries in routes.
- [x] I have not put queries in Blade templates.
- [x] I have not put JS and CSS in Blade templates and have not put any HTML in PHP classes.
- [x] I have removed all debug code.

#### Nuxt coding practices
- [ ] My Vue Components names are short, describes what it's intended use is and is in PascalCase
- [ ] All the components exclusive to the page are in the page's components folder
- [ ] I have preferred using Composition API.
- [ ] I have preferred using <script setup> tag instead of the setup().
- [ ] I have used asyncData or fetch hook (avoid mounted or created hooks) to fetch any initial data that will make the component load later than page.
- [ ] I have marked any component that does not have reactive elements as functional.
- [ ] I have used key for any v-for used.
- [ ] I have not used v-for and v-if in the same element.
- [ ] I have declared props with camelCase but used in templates with kebab-case.
- [ ] I have validated my props with good definition, and have set there default values
- [ ] I have added a Readme.md file for each component I have created.
- [x] All my page specific methods and variables are in the page's store.

---

### Feature Merge Request Checklist

- [x] have you rebased your feature with develop?
- [ ] have run npm run prod after rebase with develop branch in all your Modules & Themes? Notes, you have to reactivate modules and themes to publish assets.
- [ ] have you updated the patch (x.x.<update-this>) version config.php and composer.json of module & theme?
- [x] have you read all the comments & notes in wireframe and verified that you have taken care of that?
- [x] verify that the UI must match with design & wireframe if available?
- [x] have you applied data-wdio tag to links, forms & all form inputs like input, select?
- [x] did you verify the latest commit of develop exists in your feature branch after rebase?
- [x] have you tested the feature as per the client requirement?
- [x] did you follow the proper naming conventions in your code?
- [ ] have you commented on your code with proper explanation?

we-vikram-wri231 committed:
Updated: setup configuration and account page

we-vikram-wri231 committed:
Updated: setup configuration and account page

we-vikram-wri231 committed:
Updated: setup configuration and account page

we-vikram-wri231 committed:
Updated: setup configuration and account page

we-vikram-wri231 committed:
Updated: env app name to session prefix and session cookie

we-vikram-wri231 committed:
Updated: env app name to session prefix and session cookie

we-vikram-wri231 committed:
Updated: env app name to session prefix and session cookie

we-vikram-wri231 committed:
Updated: rebuild assets

we-vikram-wri231 committed:
Updated: rebuild assets

we-vikram-wri231 committed:
Updated: rebuild assets

we-vikram-wri231 committed:
Updated: rebuild assets

we-vikram-wri231 committed:
Updated: restricted action

we-vikram-wri231 committed:
Updated: restricted action

we-vikram-wri231 committed:
Updated: restricted action in bulk action

we-vikram-wri231 committed:
Updated: restricted action in bulk action

we-vikram-wri231 committed:
Updated: restricted action

we-vikram-wri231 committed:
Updated: restricted action

we-vikram-wri231 committed:
Fixed: permission seeds of taxonomy section

we-vikram-wri231 committed:
Fixed: permission seeds of taxonomy section

we-vikram-wri231 committed:
Updated: restricted action for bulk action

we-vikram-wri231 committed:
Updated: restricted action for bulk action

we-vikram-wri231 committed:
Updated: Active all logic

we-vikram-wri231 committed:
Updated: Active all logic

we-vikram-wri231 committed:
Fixed: role permission active all logic

we-vikram-wri231 committed:
Fixed: role permission active all logic

we-vikram-wri231 committed:
Updated: validation and placeholder text

we-vikram-wri231 committed:
Updated: validation and placeholder text

we-vikram-wri231 committed:
Updated: validation and placeholder text

we-vikram-wri231 committed:
Updated: validation and placeholder text

we-vikram-wri231 committed:
Updated: rebuild assets

we-vikram-wri231 committed:
Updated: rebuild assets

we-vikram-wri231 committed:
Updated: rebuild assets

we-vikram-wri231 committed:
Updated: rebuild assets

we-alam-a001 committed:
Updated: language string of label

we-alam-a001 committed:
Updated: language string of label

we-vikram-wri231 committed:
Merge remote-tracking branch 'origin/2.x-feature/session-fixes' into 2.x-feature/session-fixes

we-alam-a001 committed:
Updated: localization of dashboard page

we-alam-a001 committed:
Updated: sidebar menu label extend and dashboard

we-alam-a001 committed:
Updated: sidebar menu label extend and dashboard

we-alam-a001 committed:
Added: conditional if for language string

we-alam-a001 committed:
Added: conditional if for language string

we-alam-a001 committed:
Fixed: login error while language is fr

we-alam-a001 committed:
Fixed: login error while language is fr

we-alam-a001 committed:
Updated: extend controller for label

we-alam-a001 committed:
Updated: extend controller for label

we-alam-a001 committed:
Fixed: to show bold text

we-alam-a001 committed:
Fixed: to show bold text

we-alam-a001 committed:
Updated: sidebar menu label extend and dashboard 


Added: conditional if for language string


Fixed: login error while language is fr


Updated: extend controller for label


Fixed: to show bold text

we-alam-a001 committed:
Merge remote-tracking branch 'origin/2.x-feature/update-dashbaord-localization' into 2.x-feature/update-dashbaord-localization

we-alam-a001 committed:
Updated: dashboard language string

we-alam-a001 committed:
Updated: dashboard language string

we-alam-a001 committed:
Removed: unused line

we-alam-a001 committed:
Removed: unused line

we-vikram-wri231 committed:
Merge remote-tracking branch 'origin/2.x-feature/added-restricted-actions-users' into 2.x-feature/added-restricted-actions-users

we-vikram-wri231 committed:
Updated: view role bulk function

we-vikram-wri231 committed:
Updated: bulk all condition

we-vikram-wri231 committed:
Updated: roles bulk actions

we-vikram-wri231 committed:
Updated: permission denied response

we-vikram-wri231 committed:
Updated: permission denied response

we-alam-a001 committed:
Updated: registrations permission denied response

we-alam-a001 committed:
Updated: registrations permission denied response

we-vikram-wri231 committed:
Updated: haspermission param

we-vikram-wri231 committed:
Updated: haspermission param

we-alam-a001 committed:
Updated: users permission denied response

we-alam-a001 committed:
Updated: users permission denied response

we-alam-a001 committed:
Updated: all haspermission param

we-alam-a001 committed:
Updated: all haspermission param

we-alam-a001 committed:
Updated: permission denied response in roles

we-alam-a001 committed:
Updated: permission denied response in roles

we-alam-a001 committed:
Updated: permission denied response in permissions

we-alam-a001 committed:
Updated: permission denied response in permissions

we-alam-a001 committed:
Updated: permission denied response in modules

we-alam-a001 committed:
Updated: permission denied response in modules

we-alam-a001 committed:
Updated: media permission denied response

we-alam-a001 committed:
Updated: media permission denied response

we-alam-a001 committed:
Updated: taxonomies permission denied response

we-alam-a001 committed:
Updated: taxonomies permission denied response

we-alam-a001 committed:
Updated: themes permission denied response

we-alam-a001 committed:
Updated: themes permission denied response

we-alam-a001 committed:
Updated: advanced permission denied response

we-alam-a001 committed:
Updated: advanced permission denied response

we-alam-a001 committed:
Updated settings permission denied response

we-alam-a001 committed:
Updated settings permission denied response

we-alam-a001 committed:
Updated: permission denied response

we-alam-a001 committed:
Updated: permission denied response

we-alam-a001 committed:
Updated: theme controller

we-alam-a001 committed:
Updated: theme controller

we-alam-a001 committed:
Updated: permission denied response


Updated: registrations permission denied response


Updated: haspermission param


Updated: users permission denied response


Updated: all haspermission param


Updated: permission denied response in roles


Updated: permission denied response in permissions


Updated: permission denied response in modules


Updated: media permission denied response


Updated: taxonomies permission denied response


Updated: themes permission denied response


Updated: advanced permission denied response


Updated settings permission denied response


Updated: permission denied response


Updated: theme controller

we-alam-a001 committed:
Merge remote-tracking branch 'origin/2.x-feature/permission-denied-response' into 2.x-feature/permission-denied-response

themodernpk commented on Http/Controllers/Backend/Advanced/BatchesController.php:
```diff
@@ -15,11 +15,10 @@ public function __construct()
-        if (!Auth::user()->hasPermission('has-access-of-advanced-section')) {
-            $response['success'] = false;
-            $response['errors'][] = trans("vaahcms::messages.permission_denied");
+        $permission_slug = 'has-access-of-advanced-section';
```
@we-vikram-wri231 this should be array instead of line number 21, make this change everywhere else as well

themodernpk commented on Http/Controllers/Backend/Advanced/BatchesController.php:
```diff
@@ -15,11 +15,10 @@ public function __construct()
     //----------------------------------------------------------
     public function getAssets(Request $request): JsonResponse
     {
-        if (!Auth::user()->hasPermission('has-access-of-advanced-section')) {
```
@we-vikram-wri231 writing `has-access-of-advanced-section` text multiple time, no point

we-alam-a001 committed:
Added: functions to get language string

we-alam-a001 committed:
Added: functions to get language string

we-alam-a001 committed:
Added: run migrations and seeders after installed

we-alam-a001 committed:
Added: run migrations and seeders after installed

we-alam-a001 committed:
Updated: setup stores

we-alam-a001 committed:
Updated: setup stores

we-alam-a001 committed:
Updated: run seeders method

we-alam-a001 committed:
Updated: run seeders method

we-alam-a001 committed:
Updated: run migrations and seeders after installed

we-alam-a001 committed:
Updated: run migrations and seeders after installed

we-alam-a001 committed:
Updated: function in setup store

we-alam-a001 committed:
Updated: function in setup store

we-alam-a001 committed:
Added: run seeder functionality in localization

we-alam-a001 committed:
Added: run seeder functionality in localization

we-alam-a001 committed:
Added: functions to get language string


Added: run migrations and seeders after installed


Updated: setup stores


Updated: run seeders method


Updated: run migrations and seeders after installed


Updated: function in setup store


Added: run seeder functionality in localization

we-alam-a001 committed:
Merge remote-tracking branch 'origin/2.x-feature/generate-language-file-option' into 2.x-feature/generate-language-file-option

we-alam-a001 committed:
Updated: naming of language strings

we-alam-a001 committed:
Updated: naming of language strings

we-vikram-wri231 committed:
Updated: permission denied response

we-vikram-wri231 committed:
Updated: permission denied response

we-alam-a001 committed:
Updated: naming convention

we-alam-a001 committed:
Updated: naming convention

we-alam-a001 committed:
Updated: permission denied response

we-alam-a001 committed:
Updated: permission denied response

we-alam-a001 committed:
Updated: registrations permission response

we-alam-a001 committed:
Updated: registrations permission response

we-alam-a001 committed:
Updated: users permission response

we-alam-a001 committed:
Updated: users permission response

we-alam-a001 committed:
Updated: themes permission response

we-alam-a001 committed:
Updated: themes permission response

we-alam-a001 committed:
Updated: taxonomy permission response

we-alam-a001 committed:
Updated: taxonomy permission response

we-alam-a001 committed:
Updated: roles permission response

we-alam-a001 committed:
Updated: roles permission response

we-alam-a001 committed:
Updated: permission denied response of permission

we-alam-a001 committed:
Updated: permission denied response of permission

we-alam-a001 committed:
Updated: modules permission response

we-alam-a001 committed:
Updated: modules permission response

we-alam-a001 committed:
Updated: media permission response

we-alam-a001 committed:
Updated: media permission response

we-alam-a001 committed:
Updated: settings permission response

we-alam-a001 committed:
Updated: settings permission response

we-alam-a001 committed:
Updated: permission response in backup and settings

we-alam-a001 committed:
Updated: permission response in backup and settings

we-alam-a001 committed:
Updated: advanced permission response

we-alam-a001 committed:
Updated: advanced permission response

we-alam-a001 committed:
Updated: if condition in users actions

we-alam-a001 committed:
Updated: if condition in users actions

we-alam-a001 committed:
Updated: if condition in users form and view

we-alam-a001 committed:
Updated: if condition in users form and view

we-alam-a001 committed:
Updated: users assets language string

we-alam-a001 committed:
Updated: users assets language string

we-alam-a001 committed:
Updated: conditions for language string error

we-alam-a001 committed:
Updated: conditions for language string error

we-alam-a001 committed:
Updated: label of extend

we-alam-a001 committed:
Updated: label of extend

we-alam-a001 committed:
Updated: condition when no strings provided

we-alam-a001 committed:
Updated: condition when no strings provided

we-alam-a001 committed:
Added: watch for looking store assets loaded

we-alam-a001 committed:
Added: watch for looking store assets loaded

we-alam-a001 committed:
Updated: users title

we-alam-a001 committed:
Updated: users title

themodernpk committed:
Merge pull request #241 from webreinvent/2.x-feature/added-restricted-actions-users

2.x feature -> 2.x develop | added restricted actions users

we-alam-a001 committed:
Updated: dashboard language strings

we-alam-a001 committed:
Updated: dashboard language strings

we-alam-a001 committed:
Fixed: registration not found error message

we-alam-a001 committed:
Fixed: registration not found error message

we-vikram-wri231 committed:
Merge remote-tracking branch 'origin/2.x-feature/generate-language-file-option' into 2.x-feature/generate-language-file-option

we-alam-a001 committed:
Updated: page title strings

we-alam-a001 committed:
Merge remote-tracking branch 'origin/2.x-feature/generate-language-file-option' into 2.x-feature/generate-language-file-option

we-vikram-wri231 committed:
Updated: style overflow

themodernpk committed:
Merge pull request #244 from webreinvent/2.x-feature/fixx-notification-ui

2.x feature -> 2.x develop | Updated: style overflow

themodernpk committed:
Merge pull request #243 from webreinvent/2.x-feature/generate-language-file-option

2.x feature -> 2.x develop | Added: seeds & migrations and updated language strings

we-alam-a001 committed:
Merge remote-tracking branch 'origin/2.x-feature/update-dashbaord-localization' into 2.x-feature/update-dashbaord-localization

we-alam-a001 committed:
Updated: language strings after rebase

themodernpk committed:
Merge pull request #239 from webreinvent/2.x-feature/update-dashbaord-localization

2.x feature -> 2.x develop | update dashboard localization

we-vikram-wri231 committed:
Merge remote-tracking branch 'origin/2.x-feature/session-fixes' into 2.x-feature/session-fixes

themodernpk committed:
Merge pull request #233 from webreinvent/2.x-feature/session-fixes

2.x feature 2.x develop | Updated: session keys in env

we-vikram-wri231 committed:
Merge remote-tracking branch 'origin/2.x-feature/permission-denied-response' into 2.x-feature/permission-denied-response
